### PR TITLE
Reordered the bytes of a 32-bit NAL unit prefixes storing NAL unit si…

### PR DESCRIPTION
**Pull request by dkozinski**
…zes value from processor byte order to network byte order

If encoder runs on a little-endian processor, the bytes of NALU size prefix wil be reoredred
If the stack runs on a big-endian processor, there’s nothing will be done

### CODE REVIEW

**Review by dariusz-f — APPROVED** [review]
looks good

**Comment by kpchoi** [comment]
@dkozinski @dariusz-f 

Hi, I believe the syntax should be little endian, not network byte order (big endian).
It is a syntax of EVC standard, and it is defined as little endian.
Someone might think strange, because it is not 'network order', but sometimes it happens in codec standards.

And, the following including of system-level header are not good in codec library. it would be better not use the headers.
```
#if (defined(_WIN64) || defined(_WIN32)) && !defined(__GNUC__)
#include <winsock.h>
#else
#include <arpa/inet.h>
#endif

```
If you need a byte re-ordering function, you would neeed to define it as a macro.

**Review by kpchoi — CHANGES_REQUESTED** [review]
I believe that the syntax should be little endian.

**Comment by dkozinski** [comment]
> @dkozinski @dariusz-f
> 
> Hi, I believe the syntax should be little endian, not network byte order (big endian). It is a syntax of EVC standard, and it is defined as little endian. Someone might think strange, because it is not 'network order', but sometimes it happens in codec standards.
> 
> And, the following including of system-level header are not good in codec library. it would be better not use the headers.
> 
> ```
> #if (defined(_WIN64) || defined(_WIN32)) && !defined(__GNUC__)
> #include <winsock.h>
> #else
> #include <arpa/inet.h>
> #endif
> ```
> 
> If you need a byte re-ordering function, you would neeed to define it as a macro.

From our point of view it doesn't matter whether we'll use BE or LE byte order. 
The only thing that matters for us is the fact that the code is independent from the endianness of the processor.

There is no problem if we store integer values to a file and then read it on the machine that uses the same endianness.
But this causes problems when we store integer values to a file, and then we read the value in a machine that uses the opposite endianness. Due to endianness, we’ll read in reversed values that won’t make sense.

To sum up, we'll switch from BE to LE. 
I mean that we'll reorder the bytes of 4-byte NAL unit prefix from processor byte order to Little-endian byte order instead of Big-endian byte order.

processor endianness | NAL unit size prefix byte-order
-----------------------------------------------------------
LE                                | LE
-----------------------------------------------------------
BE                               | LE

Besides, we'll remove system-level headers and use appropriate macro instead.

**Comment by kpchoi** [comment]
@dkozinski @dariusz-f 

We have double-check about the syntax again.
One codec experts said it should be big endian.
So, please wait to change code.
You could make macro, but it cannot be decided whether it is little or big for now.

Please wait a little more.

**Comment by dkozinski** [comment]
> @dkozinski @dariusz-f
> 
> We have double-checked about the syntax again. One codec expert said it should be big endian. So, please wait to change code. You could make macro, but it cannot be decided whether it is little or big for now.
> 
> Please wait a little more.
We've prepared macro that will replace the htons() function calls. 
We are ready for both options. We can save 4-byte long NAL unit size using BE byte order as well as LE byte order. Just let us know about your decision.

**Comment by kpchoi** [comment]
@dkozinski @dariusz-f 

Hi, codec experts said frame size syntax should be big endian. 
Please discard the comments about little endian.
and Let's use big endian for the syntax.

**Review by kpchoi — COMMENTED** [review]
xeve library itself should be independent to endianness.
So, big or little notation is not proper inside of library.
I will change this code.
Wait a little bit.

**Comment by kpchoi** [comment]
@dkozinski @dariusz-f 

I think I finished coding for fixing nal_unit_length syntax
Please refer and review https://github.com/mpeg5/xeve/pull/40

**Comment by kpchoi** [comment]
@dkozinski 
If this PR is not required, you could close.
